File: app/src/main/java/com/channa/mobiledatausageapp/network/APIInterface.java
package com.channa.mobiledatausageapp.network;

import com.channa.mobiledatausageapp.data.response.DatastoreResponse;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface APIInterface {

    // get mobile data usage
    @GET("action/datastore_search")
    Single<DatastoreResponse> getMobileDataUsage(@Query("resource_id") String resourceId, @Query("limit") Integer limit);

}

File: app/src/main/java/com/zak/datauses/network/APIInterface.kt
package com.zak.datauses.network

import com.zak.datauses.data.response.DatastoreResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface APIInterface {
    // get mobile data usage
    @GET("action/datastore_search")
    fun getMobileDataUsage(@Query("resource_id") resourceId: String?, @Query("limit") limit: Int?): Single<DatastoreResponse?>?
}